src/main/java/com/eprosima/idl/parser/typecode/AliasTypeCode.java

```diff
@@ -132,7 +132,27 @@ public String getName()
 
     public String getScopedname()
     {
-        if (m_scope.isEmpty())
+        String scoped_name = getFullScopedname();
+
+        if (!ctx.get_template_manager().get_current_template_stgroup().is_enabled_using_explicitly_modules())
+        {
+            return scoped_name;
+        }
+
+        String current_scope = ctx.getScope();
+
+        if(current_scope.isEmpty() || !scoped_name.startsWith(current_scope + "::"))
+            return scoped_name;
```

Comment:
```suggestion
        if(current_scope.isEmpty() || !scoped_name.startsWith(current_scope + "::"))
        {
            return scoped_name;
        }
```

Please apply also to MemberedTypeCode file.

Reply:
Done in aa929cf